src/toil_scripts/lib/__init__.py

```diff
@@ -31,6 +36,17 @@ def partitions(l, partition_size):
         yield l[i:i + partition_size]
 
 
+def get_work_directory():
+    try:
+        return os.path.join(os.environ['TOIL_WORKDIR'], str(uuid4()))
+    except KeyError:
+        fallback_dir = '/mnt/ephemeral'
+        if os.path.exists(fallback_dir):
+            return os.path.join(fallback_dir, str(uuid4()))
+        else:
+            return tempfile.mkdtemp()
+
+
```

Comment:
```
def get_work_directory():
    try:
        basedir = os.environ['TOIL_WORKDIR']
    except KeyError:
        basedir = '/mnt/ephemeral'
        if not os.path.exists(basedir):
            basedir = None
    return tempfile.mkdtemp(dir=basedir)
```

is more concise, safer (since the code in `try:` can't raise a false positive) and more uniform wrt the naming of the output directory.